[Merged by Bors] - chore: add deprecated modules

adomani opened the pull request:
These modules were renamed in #28948 and #28909, and this is semi-autogenerated PR to deprecate the old module names.

These are 9 `Data.{Complex, Real}.xxx` files that have been renamed to `Analysis.{Complex, Real}.xxx`.

I had forgotten about running `lake exe mk_all` after the deprecated modules had been generated.

---
<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

For details on the "pull request lifecycle" in mathlib, please see:
https://leanprover-community.github.io/contribute/index.html

In particular, note that most reviewers will only notice your PR
if it passes the continuous integration checks.
Please ask for help on https://leanprover.zulipchat.com if needed.

To indicate co-authors, include at least one commit authored by each
co-author among the commits in the pull request. If necessary, you may 
create empty commits to indicate co-authorship, using commands like so:

git commit --author="Author Name <author@email.com>" --allow-empty -m "add Author Name as coauthor"

When merging, all the commits will be squashed into a single commit listing all co-authors.

If you are moving or deleting declarations, please include these lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Moves:
- Vector.* -> List.Vector.*
- ...

Deletions:
- Nat.bit1_add_bit1
- ...

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]

-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

adomani committed:
chore: add deprecated modules

adomani committed:
import all

grunweg commented:
Thank you for doing this! Would you like to also do #28909 and the moving of `Data/Matrix` files? :-)
maintainer merge

adomani committed:
deprecations from #28909

adomani committed:
update mk_all

adomani commented:
I added also the deprecations from #28909.  This actually uncovered a bug in the process: if a *directory* no longer exist, the old script would fail.  I fixed this now.

bryangingechen commented:
Thanks!
bors r+
